### PR TITLE
Improve help popup

### LINKED ISSUE #75623: rustdoc: confusing use of "type"
The rustdoc help says

> Prefix searches with a type followed by a colon (e.g., fn:) to restrict the search to a given type.
> 
> Accepted types are: fn, mod, struct, enum, trait, type, macro, and const.

However, in Rust lingo, `struct` or `enum` isn't a "type". `Vec<i32>` is a type. And `macro`, `trait` and `const` certainly aren't "types".

IMO this should at least be qualified to something like "item type", but I think "item kind" would be even better.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #75623.

The second commit is just a slight improvement: the help popup won't be created until someone presses "?" or ESC. Not a big improvement in itself but considering the low amount of code required, I think it was worth the shot.

r? @jyn514

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -2797,8 +2797,8 @@ function defocusSearchBar() {
 
         var infos = [
             "Prefix searches with a type followed by a colon (e.g., <code>fn:</code>) to \
-             restrict the search to a given type.",
-            "Accepted types are: <code>fn</code>, <code>mod</code>, <code>struct</code>, \
+             restrict the search to a given item kind.",
+            "Accepted item kinds are: <code>fn</code>, <code>mod</code>, <code>struct</code>, \
```
Comment (jyn514) [inline src/librustdoc/html/static/main.js:2802]: nit: we said 'item kind' on the line before so it's clear that's what we're talking about now.
```suggestion
            "Accepted kinds are: <code>fn</code>, <code>mod</code>, <code>struct</code>, \
```

**Review by jyn514 — CHANGES_REQUESTED** [review]
The second commit I'm not comfortable approving ... I don't see anything _wrong_ but I don't think the frontend is tested very well and it doesn't look _obviously_ correct. Have you tested it locally?

I guess my concern is that there's a risk we break the help popup for not much gain.

**Comment by GuillaumeGomez** [comment]
> Have you tested it locally?

Yes I did. :) You can check easily this js code if you want:

```js
function foo() {
    console.log("foo!");
    foo = function() {};
}
foo(); // will display "foo!"
foo(); // will display nothing
```

The dark magic from JS. :3

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by jyn514** [comment]
Can you add a comment [above `buildHelperPopup`] that it has to be called every time before you use it? Before it was in the global namespace and you could assume it was there. Although maybe a comment would just be to say 'go through `getHelpElement` instead'.

r=me after that.

**Comment by jyn514** [comment]
> Can you add a comment [above `buildHelperPopup`] that it has to be called every time before you use it? Before it was in the global namespace and you could assume it was there. Although maybe a comment would just be to say 'go through `getHelpElement` instead'.

We discussed this on discord - all the code is (or should be) using `getHelpElement` anyway, so nothing has changed from that perspective.

@bors r+

**Comment by Dylan-DPC** [comment]
@bors retry yield 

(this was included in a rollup that i forgot to start after creating it :P )

**Pull request merged**